core/node/groups.go

```diff
@@ -263,7 +263,7 @@ func Online(bcfg *BuildCfg, cfg *config.Config) fx.Option {
 	shouldBitswapProvide := !cfg.Experimental.StrategicProviding
 
 	return fx.Options(
-		fx.Provide(OnlineExchange(shouldBitswapProvide)),
```

Comment:
All the other functions here seem to do the config processing before throwing it into the option function. Why are we doing this one differently, too long/too many options, etc?

Reply:
Yeah. It seemed like handling it in the top-level function would spill too much subsystem-specific logic in the top-level.